### ISSUE TITLE
[mqtt] publishMQTT action still  stops functioning 

### ISSUE THREAD

**Issue by Race666** [openhab/openhab-addons#6172]
Hi,

regarding to issue #4498. I have a similar issue with the mqtt2 binding.

I use the latest docker openhab 2.4 image 
```
openhab/openhab     2.4.0-arm64-debian   8b4362c5431b        3 days ago          877MB
```
and the behaviour that publishing stops working is still the same. Publishing is not stable and not 
reliable.
The mqtt binding stops publish after a while. Subscribing of topics works as expected.

In detail. I publish all incoming knx events by the following rules
```
 // https://community.openhab.org/t/mqtt-2-5-event-bus/76938
rule "Publish commands to the event bus"
when
    // Member of PubItems received command
    Member of gPublish2MQTT received command
then
    val mqttActions = getActions("mqtt","mqtt:broker:mosquitto") // use your broker Thing ID
    mqttActions.publishMQTT("openHAB/out/"+triggeringItem.name+"/command",receivedCommand.toString)
end


rule "Publish updates to the event bus"
when
    Member of gPublish2MQTT received update
then
    // logInfo("MQTT", "Send update")
    val mqttActions = getActions("mqtt","mqtt:broker:mosquitto") // use your broker Thing ID
    mqttActions.publishMQTT("openHAB/out/"+triggeringItem.name+"/state",triggeringItem.state.toString)
end 
```
This works for a while. Then if a publishing fails once no further publishing is possible. openhab.log:
```
 06:49:52.901 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRSollTempKueche/state performed
06:49:53.016 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempKueche/state performed
06:49:53.120 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRStellwertEsszimmer/state performed
06:49:53.121 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRStellwertTempKueche/state performed
06:49:53.235 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempArbeitszimmer/state performed
06:49:53.338 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempEsszimmer/state performed
06:49:53.362 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempEsszimmer/state performed
06:49:53.441 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRSollTempArbeitszimmer/state performed
06:49:53.544 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRSollTempEsszimmer/state performed
06:49:53.726 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempWohnzimmer/state performed
06:49:53.840 [DEBUG] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstModeEsszimmer/state performed
06:49:53.945 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempWohnzimmerSuedseite/state failed!
06:49:54.081 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRSollTempWohnzimmer/state failed!
06:49:54.216 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstModeWohnzimmer/state failed!
06:49:54.362 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRStellwertWohnzimmer/state failed!
06:49:54.483 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempKinderzimmerRechts/state failed!
06:49:54.608 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstModeKinderzimmerRechts/state failed!
06:49:54.704 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRSollTempKinderzimmerRechts/state failed!
06:49:54.807 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRStellwertKinderzimmerRechts/state failed!
06:49:54.925 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstTempKinderzimmerLinks/state failed!
06:49:55.030 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRSollTempKinderzimmerLinks/state failed!
06:49:55.139 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRIstModeKinderzimmerLinks/state failed!
06:49:55.235 [WARN ] [org.eclipse.smarthome.binding.mqtt.action.MQTTActions   ] - MQTT publish to openHAB/out/RTRStellwertKinderzimmerLinks/state failed!

```
All publish attempts may fail and I have to restart openhab. It seems the binding stops working completly (maybe a raisec ondition) when a queue overflow occured (see log above, 11 knx packets within one second).

I always have to restart openhab to get it working. 

Subscripting is still working in case of publishing fails. Stromzaehler* are subscriptions. events.log:
```
06:50:25.413 [smarthome.event.ItemStateChangedEvent                   ] - StromzaehlerNutzstromKostenHeute changed from 0.61124 to 0.61208
06:50:25.425 [smarthome.event.ItemStateChangedEvent                   ] - StromzaehlerNutzstromAktuellerVerbrauch changed from 0.292 to 0.297
06:50:25.457 [smarthome.event.ItemStateChangedEvent                   ] - StromzaehlerNutzstromZeit changed from 2019-10-04T06:49:49.000+0000 to 2019-10-04T06:50:25.000+0000

```

My config. Instead of case #4498 I use a QoS of 0.
```
  "mqtt:broker:mosquitto": {
    "class": "org.eclipse.smarthome.core.thing.internal.BridgeImpl",
    "value": {
      "label": "MQTT Broker Publish",
      "channels": [
        {
          "kind": "STATE",
          "uid": {
            "segments": [
              "mqtt",
              "broker",
              "mosquitto",
              "OpenHabMQTTSubscription"
            ]
          },
          "channelTypeUID": {
            "segments": [
              "mqtt",
              "publishTrigger"
            ]
          },
          "label": "Openhab Subscription",
          "configuration": {
            "properties": {
              "stateTopic": "openHAB/in/+/command",
              "separator": "#"
            }
          },
          "properties": {},
          "defaultTags": []
        }
      ],
      "configuration": {
        "properties": {
          "publickeypin": false,
          "clientid": "paho23408234523896675",
          "lastwill_qos": 0,
          "retain": true,
          "secure": false,
          "certificatepin": false,
          "keep_alive_time": 60000,
          "password": "MySecretPW",
          "qos": 0,
          "port": 1883,
          "host": "192.168.254.154",
          "reconnect_time": 60000,
          "username": "mqttusr"
        }
      },
      "properties": {},
      "uid": {
        "segments": [
          "mqtt",
          "broker",
          "mosquitto"
        ]
      },
      "thingTypeUID": {
        "segments": [
          "mqtt",
          "broker"
        ]
      },
      "location": "Openhab"
    }
```    
Any ideas what is going wrong?
Michael

**Comment by wborn** [comment]
The fix for #4498 is not available in 2.4.0. It is part of 2.5.0.M3 so perhaps you try that version? There have been dozens of MQTT fixes since 2.4.0. 

Note that the Docker images are rebuild to include security fixes but don't contain any openHAB fixes. So the openHAB version in that image is 9+ months old.

**Comment by Race666** [comment]
Ok, will try it. Hope the transition goes smoothly...

**Comment by Race666** [comment]
Update: docker container is up since
```
fa080d62688a        openhab/openhab:2.5.0.M3-arm64-debian   "/entrypoint.sh gosu…"   31 hours ago        Up 31 hours                                                          openhab-2.5.0-M3
```
and mqtt is still working. So I think fixes which are applied to 2.5 are working.

But for beginners which are new to OpenHab this could be really frustratingly, Because they make  tests on a version 2.4 which is marked as stable but an addon does not work as expect.  Any hints during the installation that the addon have some known issues would save a lot of time and prevent user from going to another smarthome solution.

**Comment by wborn** [comment]
Thats good news! Yes it would be nice to have an overview of confirmed open/fixed issues per add-on version preferably within the UI.
